File: DaggerMVP/app/src/main/java/com/example/cpu10225/daggermvp/data/module/AppModule.java
package com.example.cpu10225.daggermvp.data.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by cpu10225 on 14/11/2017.
 */

@Module
public class AppModule {
    Application mApplication;

    public AppModule(Application mApplication) {
        this.mApplication = mApplication;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }
}

File: KotlinOnAndroid/app/src/main/java/com/example/cpu10225/kotlinonandroid/di/AppModule.kt
package com.example.cpu10225.kotlinonandroid.di

import android.app.Application
import android.content.Context
import com.example.cpu10225.kotlinonandroid.KotlinApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by cpu10225 on 05/12/2017.
 */

@Module
class AppModule(val app: KotlinApp) {
    @Provides
    @Singleton
    fun provideContext(): Context = app

    @Provides
    @Singleton
    fun provideAplication(): Application = app
}